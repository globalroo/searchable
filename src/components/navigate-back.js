import React from "react";
import { ArrowBack } from "@material-ui/icons";

export const NavigateBackContainer = ({ children }) => {
	return (
		<div
			style={styles.container}
			className="pointer-hover"
			onClick={() => window.history.go(-1)}
		>
			<ArrowBack />
			{children}
		</div>
	);
};

const styles = {
	container: {
		display: "flex",
		flexDirection: "row",
		alignItems: "flex-start",
		justifyContent: "flex-start"
	}
};
