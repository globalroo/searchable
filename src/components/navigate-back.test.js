import React from "react";
import { render, cleanup, fireEvent } from "react-testing-library";
import { NavigateBackContainer } from "./navigate-back";
import "jest-dom/extend-expect";

describe("Asset Tile tests", () => {
	afterEach(cleanup);
	it("Renders a component with a navigate back wrapper", () => {
		const { container } = render(
			<NavigateBackContainer>
				<p>Test</p>
			</NavigateBackContainer>
		);
		expect(container).toHaveTextContent("Test");
		expect(container).toMatchSnapshot();
	});
	it("Renders the example Footer with some test links", async () => {
		const { getByTestId } = render(
			<NavigateBackContainer>
				<p data-testid="clickme">Test</p>
			</NavigateBackContainer>
		);
		const spy = jest.fn();
		const safeGo = global.history.go;
		global.history.go = spy;
		await fireEvent.click(getByTestId("clickme"));
		expect(spy).toHaveBeenCalledWith(-1);
		global.history.go = safeGo;
	});
});
