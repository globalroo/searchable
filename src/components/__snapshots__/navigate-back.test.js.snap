// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset Tile tests Renders a component with a navigate back wrapper 1`] = `
<div>
  <div
    class="pointer-hover"
    style="display: flex; flex-direction: row; align-items: flex-start; justify-content: flex-start;"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root-1"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
    <p>
      Test
    </p>
  </div>
</div>
`;
